app/controllers/course/virtual_classrooms_controller.rb

```diff
@@ -16,6 +16,30 @@ def access_link #:nodoc:
     end
   end
 
+  def recorded_videos #:nodoc:
+    authorize! :manage, @virtual_classroom
+    @braincert_api_service = Course::VirtualClassroom::BraincertApiService.new(
+      @virtual_classroom, @settings
+    )
+    respond_to do |format|
+      format.json do
+        render json: @braincert_api_service.fetch_recorded_videos
+      end
+    end
+  end
+
+  def recorded_video_link #:nodoc:
+    authorize! :manage, Course::VirtualClassroom
```

Comment:
were you trying this with an admin account ? how about a normal course lectuerer ?